x-pack/plugins/security_solution/public/timelines/components/open_timeline/note_previews/index.tsx

```diff
@@ -99,74 +97,57 @@ const DeleteNoteConfirm = React.memo<{
-          appActions.deleteNote({
-            id: noteId,
-          })
-        );
-      }
-    },
-    onError: (err: string) => {
-      addError(err, { title: i18n.DELETE_NOTE_ERROR(err) });
-    },
-  });
-}
+const DeleteNoteButton = React.memo<{ noteId?: string | null; eventId?: string | null }>(
+  ({ noteId, eventId }) => {
+    const { confirmingNoteId, setConfirmingNoteId } = useContext(TimelineDataTableContext);
+    const [showModal, setShowModal] = useState(false);
+    const { mutate, isLoading } = useDeleteNote(noteId, eventId);
 
-const DeleteNoteButton = React.memo<{ noteId?: string | null }>(({ noteId }) => {
-  const [confirmingNoteId, setConfirmingNoteId] = useState<string | null | undefined>(null);
-  const { mutate, isLoading } = useDeleteNote(noteId);
+    const handleOpenDeleteModal = useCallback(() => {
+      setShowModal(true);
+      setConfirmingNoteId(noteId);
+    }, [noteId, setConfirmingNoteId]);
 
-  const handleOpenDeleteModal = useCallback(async () => {
-    setConfirmingNoteId(noteId);
-  }, [noteId]);
+    const handleCancelDelete = useCallback(() => {
+      setShowModal(false);
+      setConfirmingNoteId(null);
+    }, [setConfirmingNoteId]);
 
-  const handleCancelDelete = useCallback(() => {
-    setConfirmingNoteId(null);
-  }, []);
+    const handleConfirmDelete = useCallback(() => {
+      mutate(noteId);
+      setShowModal(false);
+      setConfirmingNoteId(null);
+    }, [mutate, noteId, setConfirmingNoteId]);
 
-  const handleConfirmDelete = useCallback(() => {
-    mutate(noteId);
-    setConfirmingNoteId(null);
-  }, [mutate, noteId]);
+    const disableDelete = useMemo(() => {
+      return isLoading || noteId == null;
+    }, [isLoading, noteId]);
 
-  const disableDelete = useMemo(() => {
-    return isLoading || noteId == null;
-  }, [isLoading, noteId]);
+    return (
+      <>
+        <EuiButtonIcon
+          title={i18n.DELETE_NOTE}
+          aria-label={i18n.DELETE_NOTE}
+          data-test-subj={'delete-note'}
+          color="text"
+          iconType="trash"
+          onClick={handleOpenDeleteModal}
+          disabled={disableDelete}
+        />
+        {
+          /* modal cannot depend only on confirmingNoteId as confirmingNoteId is part of context and once that is populated all delete
+             button modals show up with the last delete button's modal on the top. There should be a local state such as `showModal` handling that.
 
-  return (
-    <>
-      <EuiButtonIcon
-        title={i18n.DELETE_NOTE}
-        aria-label={i18n.DELETE_NOTE}
-        data-test-subj={'delete-note'}
-        color="text"
-        iconType="trash"
-        onClick={handleOpenDeleteModal}
-        disabled={disableDelete}
-      />
-      {confirmingNoteId != null && (
-        <DeleteNoteConfirm closeModal={handleCancelDelete} confirmModal={handleConfirmDelete} />
-      )}
-    </>
-  );
-});
+             Why do we need global context holding confirmingNoteId? Revisit this design
```

Comment:
@logeekal This is a very easy fix ha just change the line below from confirmingNoteId != null to confirmingNoteId === noteId. I disagree with the rest of the comment too, a local state is potentially hundreds of pieces of state, vs a string stored in redux. 0 performance overhead, trivial to debug, if in a hypothetical future we are polling the events on the page and another user deletes or edits the note while it's pending delete, it's easier to show a toast message or something, there are many benefits and 0 negatives, just need to change the logic to be what i suggested above